dali/kernels/slice/slice_flip_normalize_permute_pad_gpu.h

```diff
@@ -45,19 +53,76 @@ class SliceFlipNormalizePermutePadGpu {
     KernelRequirements req;
     ScratchpadEstimator se;
     const size_t num_samples = in.size();
-    se.add<detail::SampleDesc<Dims>>(AllocType::Host, num_samples);
-    se.add<detail::SampleDesc<Dims>>(AllocType::GPU, num_samples);
 
-    DALI_ENFORCE(args[0].mean.size() == args[0].inv_stddev.size());
-    size_t norm_args_size = args[0].mean.size();
-    if (norm_args_size > 0) {
-      se.add<float>(AllocType::Host, 2 * norm_args_size);
-      se.add<float>(AllocType::GPU,  2 * norm_args_size);
+    norm_args_size_ = -1;
+    has_channels_ = false;
+    need_normalize_ = false;
+    for (const auto& sample_args : args) {
+      if (norm_args_size_ == -1) {
+        norm_args_size_ = sample_args.mean.size();
+        need_normalize_ = norm_args_size_ > 0;
+        has_channels_ = norm_args_size_ > 1;
+      }
+
+      if (sample_args.mean.size() != norm_args_size_ ||
+          sample_args.inv_stddev.size() != norm_args_size_)
+        throw std::invalid_argument(
+            "Normalization arguments should have the same size for all the samples");
+
+      if (has_channels_) {
+        if (sample_args.channel_dim < 0 || sample_args.channel_dim >= Dims)
+          throw std::invalid_argument(
+              "Channel dim must be valid for multi-channel normalization arguments");
+        if (norm_args_size_ != sample_args.shape[sample_args.channel_dim])
+          throw std::invalid_argument(
+              "The number of normalization arguments should match the number of channels in the "
+              "output slice");
+      }
+    }
+    if (need_normalize_) {
+      se.add<float>(AllocType::Host, num_samples * norm_args_size_);
+      se.add<float>(AllocType::Host, num_samples * norm_args_size_);
+      se.add<float>(AllocType::GPU,  num_samples * norm_args_size_);
+      se.add<float>(AllocType::GPU,  num_samples * norm_args_size_);
     }
 
+    nfill_values_ = 0;
+    for (const auto& sample_args : args) {
+      if (nfill_values_ == 0) {
+        nfill_values_ = sample_args.fill_values.size();
+      } else {
+        if (nfill_values_ != sample_args.fill_values.size())
+          throw std::invalid_argument(
+              "The number of fill values should be the same for all the samples");
+      }
+    }
+    if (nfill_values_ == 0) {
+      default_fill_values_ = true;
+      nfill_values_ = norm_args_size_;
+    } else if (nfill_values_ > 1) {
```

Comment:
Do you need this additional if here? Can it be negative?

Reply:
No, but it can be == 1